domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/Shape.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.domain.entity;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class Shape {
+
+    @NotNull
+    private final String shapeId;
+    private final float shapePtLat;
+    private final float shapePtLon;
+    private final int shapePtSequence;
+
+    @Nullable
+    private final Float shapeDistTraveled;
+
+    private Shape(@NotNull final String shapeId,
+                  float shapePtLat,
+                  float shapePtLon,
+                  final int shapePtSequence,
+                  @Nullable final Float shapeDistTraveled) {
+        this.shapeId = shapeId;
+        this.shapePtLat = shapePtLat;
+        this.shapePtLon = shapePtLon;
+        this.shapePtSequence = shapePtSequence;
+        this.shapeDistTraveled = shapeDistTraveled;
+    }
+
+    @NotNull
+    public String getShapeId() {
+        return shapeId;
+    }
+
+    @NotNull
+    public Float getShapePtLat() {
+        return shapePtLat;
+    }
+
+    @NotNull
+    public Float getShapePtLon() {
+        return shapePtLon;
+    }
+
+    @NotNull
```

Comment:
GTFS shapes.txt isn't guaranteed to be ordered, so at some point we'll need to sort `Shape` objects - grouped by `shapeId`, ordered by increasing `shapePtSequence`. To support this you may want to implement `Comparable` - see https://www.geeksforgeeks.org/comparable-vs-comparator-in-java/ for an example. We can revisit this later unless you have a good idea how this entity will be used within business validation rules now.

Reply:
I keep this in mind, and will think of this implementation when adding shape object to the Gtfs Data repository.

Opening an issue on this matter though.